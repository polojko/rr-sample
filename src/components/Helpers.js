/** Helper components */

import React from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';
import ArrowUpwardIcon from '@material-ui/icons/ArrowUpward';

const Footer = () => {
    return (
        <div className='footer'>© { new Date().getFullYear()} The Elder Scrolls Legends™. All rights reserved</div>
    );
}

const Loader = () => {
    return (
        <div className='loader'>
            <CircularProgress color='inherit' />
            <div>Loading...</div>
        </div>
    );
}

const Error = () => {
    return (
        <div className='loader'>Error occured while fetching data.</div>
    );
}

const NoData = () => {
    return (
        <div className='loader'>No more data is available.</div>
    );
}

const ScrollToTop = () => {
    return (
        <div className='scroll-to-top' title='Scroll to top'>
            <ArrowUpwardIcon onClick={() => window.scrollTo({ top: 0, behavior: 'smooth' })} />
        </div>
    )
}

export { Footer, Loader, Error, NoData, ScrollToTop };