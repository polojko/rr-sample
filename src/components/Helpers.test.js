import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import { Loader, NoData, Footer, Error, ScrollToTop, SearchMessage } from './Helpers';

afterEach(cleanup);

it('Footer component renders correctly', () => {
    const footer = renderer
        .create(<Footer />)
        .toJSON();
    expect(footer).toMatchSnapshot();
});

it('Footer contains text', () => {
    render(<Footer />)
    expect(screen.getByText(/The Elder Scrolls Legendsâ„¢/i)).toBeTruthy();
});

it('Loader component renders correctly', () => {
    const loader = renderer
        .create(<Loader />)
        .toJSON();
    expect(loader).toMatchSnapshot();
});

it('Loader contains text', () => {
    render(<Loader />)
    expect(screen.getByText(/Loading.../i)).toBeTruthy();
});

it('Error component renders correctly', () => {
    const error = renderer
        .create(<Error />)
        .toJSON();
    expect(error).toMatchSnapshot();
});

it('Error contains text', () => {
    render(<Error />)
    expect(screen.getByText(/Error occured while fetching data./i)).toBeTruthy();
});

it('NoData component renders correctly', () => {
    const error = renderer
        .create(<NoData />)
        .toJSON();
    expect(error).toMatchSnapshot();
});

it('NoData contains text', () => {
    render(<NoData />)
    expect(screen.getByText(/No more data is available./i)).toBeTruthy();
});

it('ScrollToTop component renders correctly', () => {
    const scrollToTop = renderer
        .create(<ScrollToTop />)
        .toJSON();
    expect(scrollToTop).toMatchSnapshot();
});

it('SearchMessage component renders correctly', () => {
    const messageData = 'test testing';
    const message = renderer
        .create(<SearchMessage message={messageData} />)
        .toJSON();
    expect(message).toMatchSnapshot();
});

it('NoData contains text', () => {
    const messageData = 'test testing';
    render(<SearchMessage  message={messageData} />)
    expect(screen.getByText(/test testing/i)).toBeTruthy();
});
