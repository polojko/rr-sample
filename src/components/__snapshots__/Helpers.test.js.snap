// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component renders correctly 1`] = `
<div
  className="loader"
>
  Error occured while fetching data.
</div>
`;

exports[`Footer component renders correctly 1`] = `
<div
  className="footer"
>
  © 
  2021
   The Elder Scrolls Legends™. All rights reserved
</div>
`;

exports[`Loader component renders correctly 1`] = `
<div
  className="loader"
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
  <p>
    Loading...
  </p>
</div>
`;

exports[`NoData component renders correctly 1`] = `
<div
  className="loader"
>
  No more data is available.
</div>
`;

exports[`ScrollToTop component renders correctly 1`] = `
<div
  className="scroll-to-top"
  title="Scroll to top"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
    />
  </svg>
</div>
`;

exports[`SearchMessage component renders correctly 1`] = `
<div
  className="search-message"
>
  test testing
</div>
`;
